### ISSUE TITLE
Cypress does not run the same test cases for different URLs by using forEach() anymore

### ISSUE THREAD

**Issue by fpachelli** [cypress-io/cypress#5804]
### Steps to reproduce:

Try to execute the same test cases in different URLs, following the template of the official Cypress link below:

https://docs.cypress.io/faq/questions/using-cypress-faq.html#Can-I-run-the-same-tests-on-multiple-subdomains

This simple code below, copied from the link above, is enough to reproduce the problem:
```
const urls = ['https://docs.cypress.io', 'https://www.cypress.io']

describe('Logo', () => {
  urls.forEach((url) => {
    it(`Should display logo on ${url}`, () => {
      cy.visit(url)
      cy.get('#logo img')
        .should('have.attr', 'src')
        .and('include', 'logo')
    })
  })
})
```

### Current behavior:
Not possible to run the same test cases for different urls, only the last value in Array is considered (the test failed, but this is not the problem, the issue is not being run for the 2 urls)

![image](https://user-images.githubusercontent.com/53526290/69724151-0a5da880-10fa-11ea-927b-d6adb450759b.png)

**I have tried different tests in different files using the same idea with no success.**

So I tried to replace the **"forEach(url)"** with:

- **"for(let url of ulrs)"** 
- and with **"for(let url = 0; url < urls.length; url++)"** 

and none of the alternatives have worked either.

### Desired behavior:

The test cases inside the **"forEach(url)"** block should be executed once to each value inside the **"urls"** array

![image](https://user-images.githubusercontent.com/53526290/69724897-a805a780-10fb-11ea-8645-8132c1fdf560.png)

### Versions

I have tried to run this without success in two Cypress versions:

- 3.4.1
- 3.6.1 (the most recent at the moment)

**Comment by nodi1995** [comment]
It seems that this is not the problem related to forEach. When u inspect cypress.io, there is no element with logo id, like in docs.cypress.io. I do not know was that case before (maybe there was some update on cypress website)... 
cypress.io => parent of logo image is a>div.getsby-image-wrapper
docs.cypress.io => parent of logo image is a#logo
You need to read errors, and inspects websites you want to test

**Comment by fpachelli** [comment]
Hi nodi. Well, as I said in the Current Behavior of the bug: **"(the test failed, but this is not the problem, the issue is not being run for the 2 urls)"** - so yes, I know it is failing because no such element was found. I have done stuff much more complex that that in Cypress, my project is well structured.

I only used this simple example I got from the official Cypress site to be easier to illustrate, to say that even for code provided by Cypress, the problem was present.  

The problem is that I did try other variations of the idea of running the same test cases on multiple subdomains in other suites I have in my project, files that work fine and find every element in DOM, but in all cases I was only able to run only the **last** value in the array. 

Unless I create an array like that: **const urls = ['www.google.com', 'www.gmail.com', 'www.amazon.com', 'www.gmail.com']** - in this specific case, it runs for Gmail twice, because it only gets the last value in the array, but if this value was duplicated in the same array, then the tests are executed twice. This sounds like a strange bug for me.  

But yesterday (one day after I have opened this bug), the version 3.7.0 was released. It included several changes and fixes about domains and subdomains, according to the release notes. I updated my Cypress and my suites started to be run on multiple urls WITHOUT any code change. So, or it was really an issue at the last Cypress versions (that was not present at the first semester, because I found answers over internet with examples up to may, 2019 that worked according to people). Or it was any kind of corruption in my node_modules or something like that, that by recreating everything by update worked, hard to know.
